core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -123,12 +133,30 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf:SparkConf = null
+
   override def open(): BlockObjectWriter = {
     if (hasBeenClosed) {
       throw new IllegalStateException("Writer already closed. Cannot be reopened.")
     }
     fos = new FileOutputStream(file, true)
-    ts = new TimeTrackingOutputStream(fos)
+    val isEncryptedShuffle = sparkConf.getBoolean("spark.encrypted.shuffle", false)
```

Comment:
@CodingCat , thanks for your comment. I have modified https://github.com/kellyzly/spark/blob/master/core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala#L143 as following:
```
   val isEncryptedShuffle: Boolean = if (sparkConf != null)
      sparkConf.getBoolean("spark.encrypted" +
        ".shuffle",
        false)
    else
      false
```
Later will submit the code, do you think it can solve the problem you pointed out?